foo_discord_rich/discord/uploader.h

```diff
@@ -0,0 +1,32 @@
+﻿#pragma once
+
+namespace drp::uploader
+{
+
+
+class threaded_process_artwork_uploader : public threaded_process_callback
+{
+public:
+    threaded_process_artwork_uploader( const pfc::map_t<metadb_index_hash, metadb_handle_ptr>& hashes );
+    void on_init(ctx_t p_wnd) override;
+    void run(threaded_process_status & p_status,abort_callback & p_abort) override;
+    void on_done(ctx_t p_wnd,bool p_was_aborted) override;
+private:
+    pfc::map_t<metadb_index_hash, metadb_handle_ptr> hashes_{};
+};
+
+
+struct artwork_info
+{
+    album_art_data_ptr data;
```

Comment:
`album_art_data_ptr data = nullptr;` ?

Reply:
`album_art_data_ptr` is a shared pointer kind of structure so a nullptr default value would be problematic. AFAIK the default constructor is used for the default value which creates an `album_art_data_ptr` object with the internal pointer set to NULL.

Reply:
i see. looks great then! 👍